aws-predictions/src/main/java/com/amplifyframework/predictions/aws/http/LivenessWebSocket.kt

```diff
@@ -86,13 +106,31 @@ internal class LivenessWebSocket(
     private var webSocketError: PredictionsException? = null
     internal var clientStoppedSession = false
     val json = Json { ignoreUnknownKeys = true }
+    val FIVE_MINUTES = 1000 * 60 * 5
+    val datePattern = "EEE, d MMM yyyy HH:mm:ss z"
 
     @VisibleForTesting
     internal var webSocketListener = object : WebSocketListener() {
         override fun onOpen(webSocket: WebSocket, response: Response) {
             LOG.debug("WebSocket onOpen")
-            super.onOpen(webSocket, response)
-            this@LivenessWebSocket.webSocket = webSocket
+
+            // device time may be set incorrectly; read the header to skew time and retry
+            val sdf = SimpleDateFormat(datePattern, Locale.getDefault())
+            val date = response.header("Date")?.let { sdf.parse(it) }
+            val tempOffset = if (date != null) {
+                date.time - adjustedDate()
+            } else 0
+
+            // if offset is > 5 minutes, server will reject the request
+            if (kotlin.math.abs(tempOffset) < FIVE_MINUTES) {
```

Comment:
Can we also document the mechanism's reliance on session reinitialization to maintain time accuracy. Additionally, should we consider implementing a failsafe mechanism to close and reopen the connection if a session persists longer than expected or a warning that alerts when the session exceeds typical duration thresholds

Reply:
Once a session has been successfully established, it can't be re-started because a sessionId for liveness can only be used once. First point on better documenting why we are doing what we are here, would be helpful for future.